[Main]: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited

I see random test failure for PR tests on Linux 

https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_linux_pytest/2449/testReport/junit/src.tribler-core.tribler_core.components.restapi.rest.tests/test_settings_endpoint/test_set_rate_settings/

```python
Error Message
RuntimeError: 1 Runtime Warning,
/usr/lib/python3.8/weakref.py:131:coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
Stacktrace
pyfuncitem = <Function test_set_rate_settings>

    def pytest_pyfunc_call(pyfuncitem):  # type: ignore
        """
        Run coroutines in an event loop instead of a normal function call.
        """
        fast = pyfuncitem.config.getoption("--aiohttp-fast")
        if asyncio.iscoroutinefunction(pyfuncitem.function):
            existing_loop = pyfuncitem.funcargs.get('proactor_loop')\
                or pyfuncitem.funcargs.get('loop', None)
            with _runtime_warning_context():
                with _passthrough_loop_context(existing_loop, fast=fast) as _loop:
                    testargs = {arg: pyfuncitem.funcargs[arg]
                                for arg in pyfuncitem._fixtureinfo.argnames}
>                   _loop.run_until_complete(pyfuncitem.obj(**testargs))

/usr/local/lib/python3.8/dist-packages/aiohttp/pytest_plugin.py:170: 
```

<details> <summary>Full error</summary>

```python
Error Message
RuntimeError: 1 Runtime Warning,
/usr/lib/python3.8/weakref.py:131:coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
Stacktrace
pyfuncitem = <Function test_set_rate_settings>

    def pytest_pyfunc_call(pyfuncitem):  # type: ignore
        """
        Run coroutines in an event loop instead of a normal function call.
        """
        fast = pyfuncitem.config.getoption("--aiohttp-fast")
        if asyncio.iscoroutinefunction(pyfuncitem.function):
            existing_loop = pyfuncitem.funcargs.get('proactor_loop')\
                or pyfuncitem.funcargs.get('loop', None)
            with _runtime_warning_context():
                with _passthrough_loop_context(existing_loop, fast=fast) as _loop:
                    testargs = {arg: pyfuncitem.funcargs[arg]
                                for arg in pyfuncitem._fixtureinfo.argnames}
>                   _loop.run_until_complete(pyfuncitem.obj(**testargs))

/usr/local/lib/python3.8/dist-packages/aiohttp/pytest_plugin.py:170: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/contextlib.py:120: in __exit__
    next(self.gen)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    @contextlib.contextmanager
    def _runtime_warning_context():  # type: ignore
        """
        Context manager which checks for RuntimeWarnings, specifically to
        avoid "coroutine 'X' was never awaited" warnings being missed.
    
        If RuntimeWarnings occur in the context a RuntimeError is raised.
        """
        with warnings.catch_warnings(record=True) as _warnings:
            yield
            rw = ['{w.filename}:{w.lineno}:{w.message}'.format(w=w)
                  for w in _warnings  # type: ignore
                  if w.category == RuntimeWarning]
            if rw:
>               raise RuntimeError('{} Runtime Warning{},\n{}'.format(
                    len(rw),
                    '' if len(rw) == 1 else 's',
                    '\n'.join(rw)
                ))
E               RuntimeError: 1 Runtime Warning,
E               /usr/lib/python3.8/weakref.py:131:coroutine 'AsyncMockMixin._execute_mock_call' was never awaited

/usr/local/lib/python3.8/dist-packages/aiohttp/pytest_plugin.py:127: RuntimeError
```
</details>

--- a/src/tribler-core/tribler_core/components/restapi/rest/tests/test_events_endpoint.py
+++ b/src/tribler-core/tribler_core/components/restapi/rest/tests/test_events_endpoint.py
@@ -161,7 +161,7 @@ async def test_on_tribler_exception_stores_only_first_error(endpoint, reported_e
 @patch.object(EventsEndpoint, 'register_anonymous_task', new=AsyncMock(side_effect=CancelledError))
 @patch.object(RESTStreamResponse, 'prepare', new=AsyncMock())
 @patch.object(RESTStreamResponse, 'write', new_callable=AsyncMock)
-@patch.object(EventsEndpoint, 'encode_message', new_callable=AsyncMock)
+@patch.object(EventsEndpoint, 'encode_message')
 async def test_get_events_has_undelivered_error(mocked_encode_message, mocked_write, endpoint):
     # test that in case `self.undelivered_error` is not None, then it will be sent
     endpoint.undelivered_error = {'undelivered': 'error'}

